packages/bind/src/applyBindings.ts

```diff
@@ -239,31 +271,31 @@ function applyBindingsToNodeInternal (node, sourceBindings, bindingContext, asyn
               } else {
                 return valueAccessor(optionalValue)
               }
-            } : (bindingKey) => bindings[bindingKey]
+            } : (bindingKey) => bindings![bindingKey]
 
         // Use of allBindings as a function is maintained for backwards compatibility, but its use is deprecated
-    function allBindings () {
+    const allBindings: AllBindings = function () : any {
       return objectMap(bindingsUpdater ? bindingsUpdater() : bindings, evaluateValueAccessor)
     }
 
         // The following is the 3.x allBindings API
-    allBindings.has = (key) => key in bindings
-    allBindings.get = (key) => bindings[key] && evaluateValueAccessor(getValueAccessor(key))
+    allBindings.has = (key : string) => key in bindings!
+    allBindings.get = (key : string) => bindings![key] && evaluateValueAccessor(getValueAccessor(key))
 
     if (bindingEvent.childrenComplete in bindings) {
       bindingEvent.subscribe(node, bindingEvent.childrenComplete, () => {
-        const callback = evaluateValueAccessor(bindings[bindingEvent.childrenComplete])
+        const callback = evaluateValueAccessor(bindings![bindingEvent.childrenComplete])
         if (!callback) { return }
         const nodes = virtualElements.childNodes(node)
         if (nodes.length) { callback(nodes, dataFor(nodes[0])) }
-      })
+      }, null)
     }
 
     const bindingsGenerated = topologicalSortBindings(bindings, $component)
-    const nodeAsyncBindingPromises = new Set()
+    const nodeAsyncBindingPromises = new Set<Promise<any>>()
     for (const [key, BindingHandlerClass] of bindingsGenerated) {
         // Go through the sorted bindings, calling init and update for each
-      function reportBindingError (during, errorCaptured) {
+      const reportBindingError = function (during, errorCaptured) {
```

Comment:
it seems odd to make this anonymous, so it won't report in stack traces. probably ok, though.

Reply:
IMO such a function should even better be anonymous and const, otherwise there may be problems with the scope. https://stackoverflow.com/questions/750486/javascript-closure-inside-loops-simple-practical-example